src/material/datepicker/datepicker-base.ts

```diff
@@ -223,14 +231,15 @@ export interface MatDatepickerControl<D> {
   disabled: boolean;
   dateFilter: DateFilterFn<D>;
   getConnectedOverlayOrigin(): ElementRef;
-  _disabledChange: Observable<boolean>;
+  _stateChanges: Observable<void>;
 }
 
 /** Base class for a datepicker. */
 @Directive()
 export abstract class MatDatepickerBase<C extends MatDatepickerControl<D>, S,
   D = ExtractDateTypeFromSelection<S>> implements OnDestroy, CanColor, OnChanges {
   private _scrollStrategy: () => ScrollStrategy;
+  private _inputStateChanges = Subscription.EMPTY;
```

Comment:
you can create one subscription for everything that needs to be unsubscribed on destroy, then do `_subscriptions.add(...subscribe())`, or use `takeUntil(_destroyed)`, not sure where we landed on that